rootfs/etc/s6/mariadb/run

```diff
@@ -29,18 +29,17 @@ then
   echo "DROP DATABASE IF EXISTS test;" >> ${INIT_FILE}
   echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
 
-  if [[ -n "${MARIADB_USERNAME}" && -n "${MARIADB_DATABASE}" ]]
+  MARIADB_DATABASE=${MARIADB_DATABASE:-$MARIADB_USERNAME}
+  if [ ! -z "${MARIADB_DATABASE}" ]
   then
-    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_USERNAME}\`;" >> ${INIT_FILE}
-    echo "GRANT ALL PRIVILEGES ON \`${MARIADB_DATABASE}\`.* TO '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE}
-    echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
-  fi
-
-  if [[ -n "${MARIADB_USERNAME}" && -z "${MARIADB_DATABASE}" ]]
-  then
-    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_USERNAME}\`;" >> ${INIT_FILE}
-    echo "GRANT ALL PRIVILEGES ON \`${MARIADB_USERNAME}\`.* TO '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE}
-    echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
+    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_DATABASE}\`;" >> ${INIT_FILE}
+
+    if [ ! -z "${MARIADB_USERNAME}" ]
+    then
+      echo "CREATE USER IF NOT EXISTS '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE} # is able to create an User with NO Password
```

Comment:
I'm not going to put any more effort into this, because I don' thave the patience to wait a full month for anybody to incorporate 11 lines of code.

Nothing personal, I just don't have the patience.

Wish you good luck,
Regards,
Me.